testing/sdk_plan.py

```diff
@@ -136,20 +143,31 @@ def wait_for_plan_status(
     else:
         statuses = status
 
+    initial_failures = sdk_metrics.sum_service_failures(service_name)
+
     @retrying.retry(
-        wait_fixed=1000, stop_max_delay=timeout_seconds * 1000, retry_on_result=lambda res: not res
+        wait_fixed=1000,
+        stop_max_delay=timeout_seconds * 1000,
+        retry_on_result=lambda res: not res,
+        retry_on_exception=lambda e: not isinstance(e, TaskFailuresExceededException),
     )
     def fn():
+        failures = sdk_metrics.sum_service_failures(service_name)
+        if failures - initial_failures > ALLOWED_FAILURES_INCREASE:
+            log.error(
```

Comment:
Let's not use `.format()` in logging strings as these support `%s` followed by a list of variables to interpolate.